src/algorithms/MathUtil.java

```diff
@@ -38,7 +42,10 @@ public static int pow(int number, int pow){
      * @return calculated number
      */
     public static int factorial(int n) {
-       return 0;
+        if (n == 0) {
+            return 1;
+        } else
```

Comment:
You don't need **ELSE statement**, just put **return n * factorial(n - 1);** after **IF statement**